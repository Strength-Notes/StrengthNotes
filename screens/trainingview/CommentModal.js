import React from 'react';
import {
  Dimensions,
  StyleSheet,
  View,
  Text,
  TextInput,
} from 'react-native';
import Modal from 'react-native-modal';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
  centeredContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  modalContainer: {
    width: Dimensions.get('window').width * 0.85,
    margin: 20,
    backgroundColor: 'lightgray',
    borderRadius: 20,
    padding: 15,
    alignItems: 'center',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 3.84,
    elevation: 5,
  },
  header: {
    fontWeight: 'bold',
  },
});

const CommentModal = ({
  comment,
  onChangeComment,
  modalVisible,
  closeModal,
}) => (
  <Modal
    animationType="fade"
    transparent
    isVisible={modalVisible}
    onBackdropPress={closeModal}
    onBackButtonPress={closeModal}
  >
    <View style={styles.centeredContainer}>
      <View style={styles.modalContainer}>
        <Text style={styles.header}>Comment</Text>

        <View
          style={{
            backgroundColor: 'black',
            height: 0.5,
            width: styles.modalContainer.width * 0.75,
          }}
        />

        <TextInput
          style={styles.textInput}
          onChangeText={onChangeComment}
          value={comment}
          placeholder="Enter set details here"
          placeholderTextColor="gray"
          multiline
        />
      </View>
    </View>
  </Modal>
);

CommentModal.propTypes = {
  comment: PropTypes.string,
  onChangeComment: PropTypes.func.isRequired,
  modalVisible: PropTypes.bool.isRequired,
  closeModal: PropTypes.func.isRequired,
};

CommentModal.defaultProps = {
  comment: '',
};

export default CommentModal;
