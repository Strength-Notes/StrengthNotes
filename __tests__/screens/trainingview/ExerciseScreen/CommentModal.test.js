import CommentModal from '../../../../screens/trainingview/ExerciseScreen/CommentModal';

describe('<CommentModal />', () => {
  it('returns the proper component', () => {
    const component = CommentModal('Comment', jest.fn(), true, jest.fn());
    expect(component).toMatchSnapshot();
  });
});
