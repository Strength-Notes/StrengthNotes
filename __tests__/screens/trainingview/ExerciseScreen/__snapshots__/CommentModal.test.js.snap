// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentModal /> returns the proper component 1`] = `
<ReactNativeModal
  animationIn="slideInUp"
  animationInTiming={300}
  animationOut="slideOutDown"
  animationOutTiming={300}
  animationType="fade"
  avoidKeyboard={false}
  backdropColor="black"
  backdropOpacity={0.7}
  backdropTransitionInTiming={300}
  backdropTransitionOutTiming={300}
  coverScreen={true}
  customBackdrop={null}
  deviceHeight={null}
  deviceWidth={null}
  hasBackdrop={true}
  hideModalContentWhileAnimating={false}
  isVisible={false}
  onBackButtonPress={[Function]}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalShow={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  propagateSwipe={false}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  useNativeDriver={false}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "lightgray",
          "borderRadius": 20,
          "elevation": 5,
          "margin": 20,
          "padding": 15,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "width": 637.5,
        }
      }
    >
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        Comment
      </Text>
      <View
        style={
          Object {
            "backgroundColor": "black",
            "height": 0.5,
            "width": 478.125,
          }
        }
      />
      <Component
        allowFontScaling={true}
        multiline={true}
        placeholder="Enter set details here"
        placeholderTextColor="gray"
        rejectResponderTermination={true}
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
</ReactNativeModal>
`;
